### ISSUE TITLE
new-lines in XML converted to ampersands

### ISSUE THREAD

**Issue by wwwillem** [a85/POSTMan-Chrome-Extension#4405]
Postman for Mac and Linux
Version 6.0.10
OS X 10.11.6 / x64 and Fedora 27 / x64

Using PostMan to send XML requests to a cgi-bin server written in Perl. This server works fine when using curl, perl or Java as the client, but when sending a POST with PostMan the XML parser fails. It seems that PostMan modifies new-lines in the XML body into ampersands.

Here is why I think this. I created at http://ucs4.us/api/postxml.cgi a tiny server which does nothing else than returning the XML request as plain text, both in hex and ascii.

When there are no new-lines in the request ( http://ucs4.us/api/postxml-oneline.png ) everything is fine and also the Perl XML parser has no problems. However when adding some new-lines in the XML code ( http://ucs4.us/api/postxml-newline.png ), the postxml.cgi response shows that they get translated into ampersands. BTW, in this example the return code is 200 OK, but that's only because no XML parsing is happening.

When converting this request to code (I tried python and wget) also some weird conversions are happening.

**Comment by harryi3t** [comment]
@wwwillem I checked this and found that postman is not converting the new lines to ampersand.
I think the server is not able to parse the newlines correctly.

This is based on following observations, please correct me if I am wrong:

### The response from postman-echo sends '\n' in the response body
![postman](https://user-images.githubusercontent.com/5207331/38198093-fd9400ae-36a8-11e8-929b-e6f438c8d70f.png)

### Logging the actual request sent using a proxy in between (mitm)
![mitmproxy_-p_8081_--setheader___q_x-proxy__python3_6_](https://user-images.githubusercontent.com/5207331/38198222-7fdeaeba-36a9-11e8-977a-15654284a375.png)

**Comment by wwwillem** [comment]
Found the root cause of my issues and sorry for putting the blame on PostMan, shame on me!! :-) I thought it was a PostMan error because my Perl and Curl scripts were doing the job fine. However, while investigating deeper I noticed those were not inserting any newlines at all and therefore running fine.

First step was to run tcpdump on the client and I noticed that you were correct, just '0a' coming out of PostMan. Then I did the same on the server and again only '0a' coming in.

So the issue had to be either Apache, Perl, CGI or my code. I googled those three keywords and found on stackoverflow a post also about newline issues, but nothing to do with ampersands. I noticed he was using CGI a bit differently than myself.

I have always been doing
        my $qry = CGI->new(\*STDIN);
        my $req  = $qry->param("POSTDATA");
while his code was using a simpler 
        my $qry = new CGI;
        my $req  = $qry->param("POSTDATA");

Eh voila, the ampersands are suddenly gone, don't understand exactly why but simple is good <g>. I am sure I copied my code from somewhere on the web and it seemed to work. But it didn't.

Long answer, hopefully it will help in the future someone with similar problems as I had.
